transcom/mymove
# INT-B- 19933 Signed Certifications SSW

## [B-19933](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A948371)

## Summary

This PR implements backend code to capture signed certifications within the workflow on the Office side.  We will be capturing for the following use cases:

-  Pre Closeout  -- "Submit Move Details"
- Closeout  -- " confirm"

Signed certification will be captured in table signed_certifications and identified by the following new ENUMS:
 
PRE_CLOSEOUT_REVIEWED_PPM_PAYMENT
CLOSEOUT_REVIEWED_PPM_PAYMENT


## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. As a customer create and submit move containing 2 PPMs.
2. Log in as new service counselor. Review PPMs and "submit move details". Verify a record is created in table. Confirm there is only one record for PPM 1 and 2 each.
```
SELECT * FROM public.signed_certifications where certification_type = 'PRE_CLOSEOUT_REVIEWED_PPM_PAYMENT' 
and move_id = <move_id> and ppm
_id = <ppm_id>;
```
3. As customer upload PPM documentations (expenses .e.tc....) for each PPM. This will allow counselor to review and reject/approve individual expenses. Click Submit.
4. As service counselor,  click "Review Documents" for PPM 1. Review individual expenses -- select Accept/Approve. Click "confirm"; doing this will insert second record to table. Confirm there is only one record.
```
 SELECT * FROM public.signed_certifications where certification_type = 'CLOSEOUT_REVIEWED_PPM_PAYMENT' 
and move_id = <move_id> and ppm
_id = <ppm_id>;
```
5.  As service counselor,  click "Review Documents" for PPM 2. Review individual expenses -- select Reject. doing this will insert second record to table. Confirm there is only one record. Note, REJECTing one or more in step 5 will allow customer to reload any further documentations to change from REJECT to ACCEPT/APPROVE on the counselor side.
6. As customer, go into "Upload PPM documentation" for rejected PPM and resubmit again to simulate if you had uploaded new supporting documentations for counselor to ACCEPT.
7. As counselor, go into "review documents" again for PPM 2. This time ACCEPT/APPROVE for all and click confirm.  Confirm there is only one record . 

Summary: Each PPM will create two records of type PRE_CLOSEOUT_REVIEWED_PPM_PAYMENT and CLOSEOUT_REVIEWED_PPM_PAYMENT with ppm_id set. User/counselor may have to go through several iterations of uploading/reviewing before counselor ACCEPTS. No matter how many times, only 2 records must exists per PPM. We technically could  insert a new record every time counseling reviews and confirms but for now we are just creating one per touchpoint per Dan M's recommendation. We will just update existing certification for any subsequent followup review by counselor.  

### Frontend


### Backend
